command/agent/agent.go

```diff
@@ -212,6 +212,33 @@ func convertServerConfig(agentConfig *Config) (*nomad.Config, error) {
 			return nil, fmt.Errorf("raft_multiplier cannot be %d. Must be between 1 and %d", *agentConfig.Server.RaftMultiplier, MaxRaftMultiplier)
 		}
 	}
+
+	if vPtr := agentConfig.Server.RaftTrailingLogs; vPtr != nil {
+		conf.RaftConfig.TrailingLogs = uint64(*vPtr)
+		if *vPtr < 1 {
+			return nil, fmt.Errorf("raft_trailing_logs must be non-negative, got %d", *vPtr)
+		}
```

Comment:
It's maybe safe because we're returning an error which the caller is going to bubble up and then bail out... but we probably shouldn't do the cast and update until after we've validated `vPtr`'s value, otherwise we'll get an underflow on the config value:

```suggestion
		if *vPtr < 1 {
			return nil, fmt.Errorf("raft_trailing_logs must be non-negative, got %d", *vPtr)
		}
		conf.RaftConfig.TrailingLogs = uint64(*vPtr)
```

Reply:
Moved in both cases I saw it.  Added agent test to check the -1 cases for trailing logs and snapshot threshold.